assets/src/edit-story/components/dropzone/canvasDropZone.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+import StoryPropTypes from '../../types';
+import { useUploader } from '../../app/uploader';
+import { useMedia } from '../../app/media';
+import { disableDefaults } from './utils';
+import useDropZone from './useDropZone';
+import { DropzoneComponent, OverContent, OverlayWrapper } from './shared';
+
+function CanvasDropzone({ children }) {
+  const {
+    actions: { setIsDragging },
+    state: { isDragging },
+  } = useDropZone();
+  const { uploadFile } = useUploader();
+  const {
+    state: { DEFAULT_WIDTH },
+    actions: { insertMediaElement },
+  } = useMedia();
+
+  const onDropHandler = (evt) => {
```

Comment:
I guess the PR is still WIP in terms of not hijacking the other DropZones and still allowing DnD for other elements as well, e.g. reordering in page carousel?

Reply:
Have you had any thoughts on the best way to do this?

Reply:
Maybe when dragging a component from the editor, some data could be passed/set with it, and in this case we could separate draggd media that's coming from outside the editor (without the set data) vs the components that are being dragged within the editor (with the set data). The Media Uploader dropzones could then only be displayed based on that. Haven't looked into it thoroughly though :)

Reply:
/cc @barklund 

I think the page carousel is likely to move toward the drag appointer we had to take for layers. Morten could you confirm?

Reply:
If that's the case then perhaps `DropZoneProvider` will be obsolete after all.

@dvoytenko Curious to know, what was the reason for having to use the drag appointer for layers?

Reply:
@miina, @barklund and I had a review with Sam. Based on the UX, and we judged that native DND wasn't a good fit. I think the main issue was lock to axis and they didn't actually want the ghost to be shown. I'm not 100%, but it seemed like similar nuances could apply to page carousel, and maybe even the grid.